gobblin-runtime/src/main/java/org/apache/gobblin/scheduler/JobScheduler.java

```diff
@@ -230,6 +236,57 @@ public void scheduleJob(Properties jobProps, JobListener jobListener)
     }
   }
 
+  public Future submitJob(Properties jobProps, JobListener jobListener, JobLauncher jobLauncher) {
+    Runnable runnable = new Runnable() {
+      @Override
+      public void run() {
+        try {
+          runJob(jobProps, jobListener, jobLauncher);
+        } catch (JobException je) {
+          LOG.error("Failed to run job " + jobProps.getProperty(ConfigurationKeys.JOB_NAME_KEY), je);
+        }
+      }
+    };
+    final Future<?> future = this.jobExecutor.submit(runnable);
+    this.submitRunnableToExecutor(runnable);
```

Comment:
What are you trying to do with line 250 and line 251? Do you need both or line 250 is good enough?